### PR TITLE
[GLUTEN-1296][VL][CI]fix: Speed up on running Spark unit tests

### PR DESCRIPTION
**Pull request by zhouyuan**
## What changes were proposed in this pull request?

This patch picked up those slow tests from Spark unit tests and run them in a parallel job to speed up the CI.

- GlutenSQLQueryTestSuite
- GlutenOrcV1SchemaPruningSuite
- GlutenOrcV2SchemaPruningSuite
- GlutenParquetV1FilterSuite
- GlutenParquetV2FilterSuite
- GlutenParquetV1SchemaPruningSuite
- GlutenParquetV2SchemaPruningSuite

related: #1296 

## How was this patch tested?

pass GHA

### CODE REVIEW

**Comment by kyligence-git** [comment]
Can one of the admins verify this patch?

**Comment by zhouyuan** [comment]
looks like this could speed up the 20.04 tests by 53 minutes
|before | after|
|--------|------|
|image| image |

File: .github/workflows/velox_be.yml [hunk 1]
```diff
@@ -131,6 +131,55 @@ jobs:
...
+          tools/gluten-te/ubuntu/cbash.sh sleep 14400
+      - name: Setup maven cache
+        run: |
+          docker cp ~/.m2/repository velox-backend-ubuntu2004-test-slow-$GITHUB_RUN_ID:/root/.m2/
+      - name: Build Gluten Velox-backend third party
+        run: |
+          docker exec velox-backend-ubuntu2004-test-slow-$GITHUB_RUN_ID bash -c '
+          cd /opt/gluten/ep/build-arrow/src && \
+          ./get_arrow.sh  && \
+          ./build_arrow.sh --build_tests=ON --build_benchmarks=ON && \
```
Comment (jinchengchenghh) [inline .github/workflows/velox_be.yml:153]: `--build_tests=ON --build_benchmarks=ON` is not needed

**Comment by jinchengchenghh** [comment]
Move TOCH and TPCDS to `velox-backend-ubuntu2004-test-slow` can save 20 min

**Comment by zhouyuan** [comment]
Here's the new run time for velox unit tests 
image

File: .github/workflows/velox_be.yml (hunk 1 above)
Comment (zhouyuan) [reply .github/workflows/velox_be.yml:128]: done

File: .github/workflows/velox_be.yml [hunk 2]
```diff
@@ -131,6 +131,55 @@ jobs:
...
+          cd /opt/gluten/ep/build-velox/src && \
+          ./get_velox.sh && \
+          ./build_velox.sh'
+      - name: Build Gluten CPP library
+        run: |
+          docker exec velox-backend-ubuntu2004-test-slow-$GITHUB_RUN_ID bash -c '
+          cd /opt/gluten/cpp && \
+          mkdir build && cd build && cmake -DBUILD_TESTS=ON -DBUILD_BENCHMARKS=ON .. && make -j'
+      - name: Build for Spark 3.2.0
```
[the remark this answers is not in the document]
Comment (zhouyuan) [reply .github/workflows/velox_be.yml:137]: seems like the time on spark 320/321 build are very short

**Review by zhztheplayer — APPROVED** [review]

**Comment by zhouyuan** [comment]
@jinchengchenghh @zhztheplayer  thanks for the review.

**Pull request merged**